pandas/io/html.py

```diff
@@ -922,6 +923,7 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
     return ret
 
 
+@deprecate_nonkeyword_arguments(version="1.4")
```

Comment:
```suggestion
@deprecate_nonkeyword_arguments(version="1.4", allowed_args=["io"])
```

? 
I know it is not needed here (since it is inferred from the signature), but I think the deprecation message is better when this is specified (saying "the first").

Reply:
Hmm, that would be reasonable if keep whatsnew file unchanged.
https://github.com/pandas-dev/pandas/pull/27573#discussion_r370082289

I think it's better to say either "all arguments buy `io`" or "all arguments but the first one" in both places.